addons/mail/static/src/core_ui/message.js

```diff
@@ -279,6 +284,29 @@ export class Message extends Component {
         this.state.isClicked = null;
     }
 
+    useReactionViewer() {
+        const message = this.message;
+        function open(ev) {
+            if (!message || message.reactions.length === 0) {
+                return;
+            }
+            ev.preventDefault();
+            const reactionMenu = registry.category("main_components").content[message.id];
+            if (!reactionMenu) {
+                registry.category("main_components").add(message.id, {
+                    Component: MessageReactionMenu,
+                    props: { close, message },
+                });
+            }
+        }
+
+        function close() {
+            registry.category("main_components").remove(message.id);
+        }
+        onWillDestroy(close);
+        return { open, close };
+    }
```

Comment:
It's best to use a `Dialog` rather than a `main_component`.
The content is simply your component.
The darken background is built-in in dialog. And the sizing is managed by props `size`, which allow consistent sizing of dialogs across Odoo.